apps/src/javalab/redux/javalabRedux.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * Redux store for generic Java Lab state.
+ */
+
+import {PayloadAction, createSlice} from '@reduxjs/toolkit';
+
+interface JavalabState {
+  isRunning: boolean;
+  isTesting: boolean;
+  backpackEnabled: boolean;
+  isStartMode: boolean;
+  levelName: string | undefined;
```

Comment:
IMO `string | undefined` might be clearer just so you have to explicitly assign `levelName` to undefined in the initial state, rather than omitting it?

Reply:
yeah I think I'll leave it as-is so it's defined in `initialState`